src/client_backend/dynamic_grpc/dynamic_grpc_client.h

```diff
@@ -0,0 +1,236 @@
+  /// \param inputs The vector of InferInput describing the model inputs.
+  /// \param outputs Optional vector of InferRequestedOutput describing how the
+  /// output must be returned. If not provided then all the outputs in the model
+  /// config will be returned as default settings.
+  /// \param headers Optional map specifying additional HTTP headers to include
+  /// in the metadata of gRPC request.
+  /// \param compression_algorithm The compression algorithm to be used
+  /// by gRPC when sending requests. By default compression is not used.
+  /// \return Error object indicating success or failure of the
+  /// request.
+  Error BidiStreamRPC(
+      InferResult** result, const InferOptions& options,
+      const std::vector<InferInput*>& inputs,
+      const std::vector<const InferRequestedOutput*>& outputs =
+          std::vector<const InferRequestedOutput*>(),
+      const Headers& headers = Headers(),
+      grpc_compression_algorithm compression_algorithm = GRPC_COMPRESS_NONE);
+
+  /// Starts a grpc bi-directional stream to send streaming inferences.
+  /// \param callback The callback function to be invoked on receiving a
+  /// response at the stream.
+  /// \param enable_stats Indicates whether client library should record the
+  /// the client-side statistics for inference requests on stream or not.
+  /// The library does not support client side statistics for decoupled
+  /// streaming. Set this option false when there is no 1:1 mapping between
+  /// request and response on the stream.
+  /// \param stream_timeout Specifies the end-to-end timeout for the streaming
+  /// connection in microseconds. The default value is 0 which means that
+  /// there is no limitation on deadline. The stream will be closed once
+  /// the specified time elapses.
+  /// \param headers Optional map specifying additional HTTP headers to
+  /// include in the metadata of gRPC request.
+  /// \param compression_algorithm The compression algorithm to be used
+  /// by gRPC when sending requests. By default compression is not used.
+  /// \return Error object indicating success or failure of the request.
+  Error StartStream(
+      OnCompleteFn callback, bool enable_stats = true,
+      uint32_t stream_timeout = 0, const Headers& headers = Headers(),
+      grpc_compression_algorithm compression_algorithm = GRPC_COMPRESS_NONE);
+
+  /// Stops an active grpc bi-directional stream, if one available.
+  /// \return Error object indicating success or failure of the request.
+  Error StopStream();
+
+  /// Returns the inference statistics of the client.
+  const InferStat& ClientInferStat() { return infer_stat_; }
+
+ protected:
+  // Update the infer stat with the given timer
+  Error UpdateInferStat(const tc::RequestTimers& timer);
+  // Enables verbose operation in the client.
+  bool verbose_;
+
+  // The inference statistic of the current client
+  InferStat infer_stat_;
+
+ private:
+  Error PreRunProcessing(
```

Comment:
Nit: this isn't inheriting from an interface, there's no technical need to have a function with this name.

I don't really understand the original use of this that's why I'm slightly pushing back on continuing its use.